hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmDBAccessIdInfo.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.ozone.om.helpers;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.hdds.StringUtils;
+
+/**
+ * This class is used for storing Ozone tenant accessId info.
+ */
+public final class OmDBAccessIdInfo {
+  /**
+   * Name of the tenant.
+   */
+  private final String tenantId;
+  /**
+   * Kerberos principal this accessId belongs to.
+   */
+  private final String kerberosPrincipal;
+  /**
+   * Shared secret of the accessId. TODO: Encryption?
+   */
+  private final String sharedSecret;
+
+  // This implies above String fields should NOT contain the split key.
+  public static final String SERIALIZATION_SPLIT_KEY = ";";
+
+  public OmDBAccessIdInfo(String tenantId,
+      String kerberosPrincipal, String sharedSecret) {
+    this.tenantId = tenantId;
+    this.kerberosPrincipal = kerberosPrincipal;
+    this.sharedSecret = sharedSecret;
+  }
+
+  private OmDBAccessIdInfo(String accessIdInfoString) {
+    String[] tInfo = accessIdInfoString.split(SERIALIZATION_SPLIT_KEY);
+    Preconditions.checkState(tInfo.length == 3,
+        "Incorrect accessIdInfoString");
+
+    tenantId = tInfo[0];
+    kerberosPrincipal = tInfo[1];
+    sharedSecret = tInfo[2];
+  }
+
+  public String getTenantId() {
+    return tenantId;
+  }
+
+  private String serialize() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(tenantId).append(SERIALIZATION_SPLIT_KEY);
+    sb.append(kerberosPrincipal).append(SERIALIZATION_SPLIT_KEY);
+    sb.append(sharedSecret);
+    return sb.toString();
+  }
+
+  /**
+   * Convert OmDBAccessIdInfo to byteArray to be persisted to DB.
+   * @return byte[]
+   */
+  public byte[] convertToByteArray() {
+    return StringUtils.string2Bytes(serialize());
```

Comment:
i understand this is the same pattern used elsewhere, but i wonder is it memory efficient. It might save some by skipping constructing the String object.
(fwiw i don't see StringUtils.string2Bytes() nor StringUtils.bytes2String() popping up in profiler yet)

Reply:
Yes it copies the string. Underneath the call it uses `str.getBytes()`, we can't seem to circumvent that if we do want to convert String to byte[]. If there is we can improve StringUtils.